### ISSUE TITLE
[BUG]: Segfault when trying to use numpy

### ISSUE THREAD

**Issue by Mec-iS** [modularml/mojo#642]
### Bug description

I have developed a small test example to try to use `numpy` in Mojo. When I try to run the script:
```
[41606:41606:20230908,151018.794358:ERROR file_io_posix.cc:144] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
[41606:41606:20230908,151018.794467:ERROR file_io_posix.cc:144] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
Stack dump:
0.      Program arguments: mojo hello.mojo
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  mojo                 0x000056106e36e957
1  mojo                 0x000056106e36c52e
2  mojo                 0x000056106e36f02f
3  libc.so.6            0x00007f4128857520
4  libpython3.10.so.1.0 0x00007f411435208e PyObject_GetAttrString + 14
5  libpython3.10.so.1.0 0x00007f408c003df7 PyObject_GetAttrString + 18446744071424384375
Segmentation fault
```

### Steps to reproduce

```mojo
from python import Python

var np: PythonObject = None

fn load():
    try:
        # this doesn't compile outside of a try block.
        # usage of try-except blocks should be documented in the Mojo doc
        # as their usage is enforced by the compiler
        np = Python.import_module("numpy")
    except:
        pass

fn main():
    np_test()

fn np_test():
    var ar: PythonObject = None
    try:
        ar = np.arange(15).reshape(3, 5)
    except:
        pass
    print(ar)
    try:
        print(ar.shape)
    except:
        pass

```
I assumed PythonObject was the right type to handle the `numpy` module and the `np.ndarray` but I am not sure. Anyway I suppose this should return an error and not crash.

### System information

```shell
I am using Ubuntu22.04LTS on x64 architecture.

`modular 0.1.4 (6b54d308)`
```

**Comment by willghatch** [comment]
For what it's worth, I just tried to repro this on MacOS and had no crash.  So this may be specific to Linux or the environment of the SDK release.

**Comment by willghatch** [comment]
Now that I try it on Linux I'm also seeing a segfault.

**Comment by austinorr** [comment]
don't forget to call your `load()` function somewhere.

**Comment by austinorr** [comment]
this worked for me on linux:
```
from python import Python

var np: PythonObject = None


fn load():
    try:
        # this doesn't compile outside of a try block.
        # usage of try-except blocks should be documented in the Mojo doc
        # as their usage is enforced by the compiler
        np = Python.import_module("numpy")
    except:
        pass


fn main():
    load()
    np_test()


fn np_test():
    var ar: PythonObject = None
    try:
        ar = np.arange(15).reshape(3, 5)
        print(ar)
        print(ar.shape)
    except:
        pass

```

**Comment by austinorr** [comment]
I use virtual envs and ran into this issue. Maybe this other [comment](https://github.com/modularml/mojo/issues/551#issuecomment-1712574082) will help too.

**Comment by Mec-iS** [comment]
> this worked for me on linux:
> 
> ```
> from python import Python
> ... 
>
> ```

Still same segfault for me. It happens as soon as `np_test` is called, after `load`

**Comment by austinorr** [comment]
@Mec-iS do you have MOJO_PYTHON_LIBRARY explicitly set?

**Comment by sontanon** [comment]
I struggled for several hours with Ubuntu 20 under WSL2 and Anaconda Python 3.11 with similar error messages for `scaling_cur_freq` and `scaling_max_freq`. 

I *reverted to 3.10* by creating a new environment with the ML bare minimums:

```bash
 conda create -n "mojo" python=3.10 pip numpy jupyter pandas matplotlib scikit-learn
```

Then, *with the environment active* (`conda activate mojo`), I reinstalled mojo (setting up `MODULAR_HOME` and appending it to my `PATH` as the installer recommends).

```bash
modular clean 
modular install mojo
```

The resulting installation has this line in my `$MODULAR_HOME/modular.cfg`:

```cfg
python_lib = /home/sontanon/anaconda3/envs/mojo/lib/libpython3.10.so.1.0
```

With this, I was able to run all Python examples and the `np_test` above. No additional setup for `MOJO_PYTHON_LIBRARY` was necessary.

To be fair, the [Get started](https://docs.modular.com/mojo/manual/get-started/index.html) says Python is only supported from 3.8 - 3.10.
